src/gf.c

```diff
@@ -994,7 +994,9 @@ JL_DLLEXPORT int jl_isa_compileable_sig(
         jl_methtable_t *kwmt = mt == jl_kwcall_mt ? jl_kwmethod_table_for(decl) : mt;
         if ((jl_value_t*)mt != jl_nothing) {
             // try to refine estimate of min and max
-            if (kwmt != NULL && kwmt != jl_type_type_mt && kwmt != jl_nonfunction_mt && kwmt != jl_kwcall_mt)
+            if (definition->max_varargs != UINT8_MAX)
+                nspec_max = nspec_min = nargs + 1 + definition->max_varargs;
```

Comment:
```suggestion
                nspec_max = nspec_min = nargs - 1 + definition->max_varargs;
```

Reply:
Can you explain the reasoning for me?

Reply:
Both of these were incorrect in the end, as far as I can tell

I think I finally got it right now 🤞